cmd/quota.go

```diff
@@ -0,0 +1,116 @@
+		if err != nil {
+			log.Error("Wrong cpu argument: " + err.Error())
+			return
+		}
+		quota.cpu = quantity
+	}
+
+	if len(quota.memoryArg) != 0 {
+		quantity, err := k8sResource.ParseQuantity(quota.memoryArg)
+		if err != nil {
+			log.Error("Wrong cpu argument: " + err.Error())
+			return
+		}
+		quota.memory = quantity
+	}
+}
+
+func checkQuota(container Container, quota quotaFlags, result *Result) {
+	if container.Resources.Limits == nil {
+		occ := Occurrence{id: ErrorResourcesLimitsNIL, kind: Warn, message: "Quota not set, please set it!"}
+		result.Occurrences = append(result.Occurrences, occ)
+		return
+	}
+
+	checkCPULimit(container, quota, result)
+	checkMemoryLimit(container, quota, result)
+}
+
+func checkCPULimit(container Container, quota quotaFlags, result *Result) {
+	cpu := container.Resources.Limits.Cpu()
+	if cpu == nil || cpu.IsZero() {
+		occ := Occurrence{id: ErrorResourcesLimitsCpuNIL, kind: Warn, message: "CPU quota not set, please set it!"}
+		result.Occurrences = append(result.Occurrences, occ)
+		return
+	}
+
+	if quota.cpu.MilliValue() > 0 && cpu.MilliValue() > quota.cpu.MilliValue() {
+		occ := Occurrence{id: ErrorResourcesLimitsCpuExceeded, kind: Warn, message: "CPU limit exceeded, please adjust it!"}
+		result.Occurrences = append(result.Occurrences, occ)
+	}
+}
+
+func checkMemoryLimit(container Container, quota quotaFlags, result *Result) {
+	memory := container.Resources.Limits.Memory()
+	if memory == nil || memory.IsZero() {
+		occ := Occurrence{id: ErrorResourcesLimitsMemoryNIL, kind: Warn, message: "Memory quota not set, please set it!"}
+		result.Occurrences = append(result.Occurrences, occ)
+		return
+	}
+
+	if quota.memory.Value() > 0 && memory.Value() > quota.memory.Value() {
+		occ := Occurrence{id: ErrorResourcesLimitsMemoryExceeded, kind: Warn, message: "Memory limit exceeded, please adjust it!"}
+		result.Occurrences = append(result.Occurrences, occ)
+	}
+}
+
+func auditQuota(quota quotaFlags, items Items) (results []Result) {
+	quota.parseQuotaFlags()
+
+	for _, item := range items.Iter() {
+		containers, result := containerIter(item)
+		for _, container := range containers {
+			checkQuota(container, quota, result)
+			if result != nil && len(result.Occurrences) > 0 {
+				results = append(results, *result)
+				break
+			}
+		}
+	}
+	for _, result := range results {
+		result.Print()
+	}
+	return
+}
+
+var quotaCmd = &cobra.Command{
+	Use:   "quota",
+	Short: "Audit containers running with quota",
+	Long: `This command determines which containers in a kubernetes cluster
+have cpu and memory quota limits defined.
+
+A PASS is given when a container has cpu and memory quota limits
+A FAIL is given when a container does not have cpu and memory quota limits
+
+Example usage:
+kubeaudit quota
+kubeaudit quota --cpu 500m --memory 256Mi`,
+	Run: runAudit(auditQuota),
+}
+
+func init() {
+	RootCmd.AddCommand(quotaCmd)
+	quotaCmd.Flags().StringVar(&quotaConfig.cpuArg, "cpu", "", "max cpu limit")
```

Comment:
Could you move them into the root command: https://github.com/Shopify/kubeaudit/blob/master/cmd/root.go as well.

Reply:
@klautcomputing but cpu and memory limits are not root flags like other flags in root.go. They are bind to quota command. So I am not sure about moving cpu and mem flags to root.go?

Reply:
I'm not sure for that too. I can reintegrate it into quota command if it's necessary. Let me know.